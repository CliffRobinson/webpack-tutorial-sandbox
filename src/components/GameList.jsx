import log from 'loglevel'
import React, { useEffect } from 'react'

import {PENDING} from '../../constants/gameStatus'
import {GameListItem} from './GameListItem'

const styleObj = {
    backgroundColor: 'white',
    border: "2px solid black"
}

export function GameList(props) {
    const {requestGamesByStatus} = props
    
    useEffect(() => {
        log.debug(`gameList component is dispatching requestGamesByStatus(pending)`)
        requestGamesByStatus(PENDING)
    }, [])

    const pendingGames = props.games[PENDING];

    return (
        <div style={styleObj}>
            <h3>This is a list of games</h3>
            {pendingGames.map((game, i) => (<GameListItem key={i} game={game}/>))}
        </div>
)
}