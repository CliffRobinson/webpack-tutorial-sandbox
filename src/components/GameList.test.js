import React from 'react'
import {shallow, mount} from 'enzyme'

import {GameList} from './GameList'

test('snapshot', () => {
    const wrapper = shallow(<GameList />)

    expect(wrapper).toMatchSnapshot()
})

test('GameList calls requestGameByStatus(pending) when it loads', () => {
    const fakeRequestGames = jest.fn()

    const wrapper = mount(<GameList requestGamesByStatus={fakeRequestGames} />)

    expect(fakeRequestGames).toBeCalled()
})