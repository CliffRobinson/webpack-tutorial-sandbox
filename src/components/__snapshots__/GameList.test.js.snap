// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot, game 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid black",
    }
  }
>
  <h3>
    This is a list of games
  </h3>
  <GameListItem
    game={
      Object {
        "id": 1,
        "name": "Harry's Hangout",
        "password": "",
        "status": "pending",
      }
    }
    key="0"
  />
  <GameListItem
    game={
      Object {
        "id": 3,
        "name": "Riddle's Ruckus",
        "password": "pureblood",
        "status": "playing",
      }
    }
    key="1"
  />
</div>
`;

exports[`snapshot, no games 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid black",
    }
  }
>
  <h3>
    This is a list of games
  </h3>
</div>
`;
